mobile/lib/modules/activities/views/activities_page.dart

```diff
@@ -1,307 +1,99 @@
     super.key,
   });
 
   @override
   Widget build(BuildContext context, WidgetRef ref) {
-    final provider =
-        activityStateProvider((albumId: albumId, assetId: assetId));
-    final activities = ref.watch(provider);
-    final inputController = useTextEditingController();
-    final inputFocusNode = useFocusNode();
-    final listViewScrollController = useScrollController();
-    final currentUser = Store.tryGet(StoreKey.currentUser);
-
-    useEffect(
-      () {
-        inputFocusNode.requestFocus();
-        return null;
-      },
-      [],
-    );
-
-    buildTitleWithTimestamp(Activity activity, {bool leftAlign = true}) {
-      final textColor = context.isDarkTheme ? Colors.white : Colors.black;
-      final textStyle = context.textTheme.bodyMedium
-          ?.copyWith(color: textColor.withOpacity(0.6));
+    // Album has to be set in the provider before reaching this page
+    final album = ref.watch(currentAlbumProvider)!;
+    final asset = ref.watch(currentAssetProvider);
+    final user = ref.watch(currentUserProvider);
 
-      return Row(
-        mainAxisAlignment: leftAlign
-            ? MainAxisAlignment.start
-            : MainAxisAlignment.spaceBetween,
-        mainAxisSize: leftAlign ? MainAxisSize.min : MainAxisSize.max,
-        children: [
-          Text(
-            activity.user.name,
-            style: textStyle,
-            overflow: TextOverflow.ellipsis,
-          ),
-          if (leftAlign)
-            Text(
-              " • ",
-              style: textStyle,
-            ),
-          Expanded(
-            child: Text(
-              activity.createdAt.copyWith().timeAgo(),
-              style: textStyle,
-              overflow: TextOverflow.ellipsis,
-              textAlign: leftAlign ? TextAlign.left : TextAlign.right,
-            ),
-          ),
-        ],
-      );
-    }
-
-    buildAssetThumbnail(Activity activity) {
-      return withAssetThumbs && activity.assetId != null
-          ? Container(
-              width: 40,
-              height: 30,
-              decoration: BoxDecoration(
-                borderRadius: const BorderRadius.all(Radius.circular(4)),
-                image: DecorationImage(
-                  image: CachedNetworkImageProvider(
-                    getThumbnailUrlForRemoteId(
-                      activity.assetId!,
-                    ),
-                    cacheKey: getThumbnailCacheKeyForRemoteId(
-                      activity.assetId!,
-                    ),
-                    headers: {
-                      "Authorization":
-                          'Bearer ${Store.get(StoreKey.accessToken)}',
-                    },
-                  ),
-                  fit: BoxFit.cover,
-                ),
-              ),
-              child: const SizedBox.shrink(),
-            )
-          : null;
-    }
+    final activityNotifier = ref
+        .read(albumActivityProvider(album.remoteId!, asset?.remoteId).notifier);
```

Comment:
Should this be `watch` as well?

Reply:
I assigned the notifier class to a variable so that it'll be easier to call the side-effects later instead of doing `ref.read(albumActivityProvider(album.remoteId!, asset?.remoteId).notifier)` whenever we need the side-effect. Since this is not the actual state but the notifier class, we do not have to watch the notifier for changes